cmake/developer_package/features.cmake

```diff
@@ -42,24 +42,31 @@ ie_dependent_option (ENABLE_COVERAGE "enable code coverage" OFF "CMAKE_CXX_COMPI
 
 # Defines CPU capabilities
 
-ie_dependent_option (ENABLE_SSE42 "Enable SSE4.2 optimizations" ON "X86_64 OR X86" OFF)
+ie_dependent_option (ENABLE_SSE42 "Enable SSE4.2 optimizations" ON "X86_64 OR (X86 AND NOT EMSCRIPTEN)" OFF)
 
-ie_dependent_option (ENABLE_AVX2 "Enable AVX2 optimizations" ON "X86_64 OR X86" OFF)
+ie_dependent_option (ENABLE_AVX2 "Enable AVX2 optimizations" ON "X86_64 OR (X86 AND NOT EMSCRIPTEN)" OFF)
 
-ie_dependent_option (ENABLE_AVX512F "Enable AVX512 optimizations" ON "X86_64 OR X86" OFF)
+ie_dependent_option (ENABLE_AVX512F "Enable AVX512 optimizations" ON "X86_64 OR (X86 AND NOT EMSCRIPTEN)" OFF)
 
-ie_option (BUILD_SHARED_LIBS "Build as a shared library" ON)
+if(EMSCRIPTEN)
+    set (BUILD_SHARED_LIBS_DEFAULT OFF)
+else()
+    set (BUILD_SHARED_LIBS_DEFAULT ON)
+endif()
+
+# Type of build, we add this as an explicit option to default it to ON
+ie_option (BUILD_SHARED_LIBS "Build as a shared library" ${BUILD_SHARED_LIBS_DEFAULT})
```

Comment:
Why not `ie_dependent_option`?

Reply:
Because it works differently, it totally disables option, while I want to change the default value